### PR TITLE
Rewrite custom element

### PR DESCRIPTION
**Pull request by esdoppio**
@hansottowirtz hasn't been responsive on #64 for quite a while, so I decided to take over.

### CODE REVIEW

**Comment by esdoppio** [comment]
Ready to 🚀

File: grammars/css.cson [hunk 1]
```diff
@@ -671,6 +667,10 @@
         'match': '(?i)(\\[)\\s*(-?[_a-z\\\\[[:^ascii:]]][_a-z0-9\\-\\\\[[:^ascii:]]]*)(?:\\s*([~|^$*]?=)\\s*(?:(-?[_a-z\\\\[[:^ascii:]]][_a-z0-9\\-\\\\[[:^ascii:]]]*)|((?>([\'"])(?:[^\\\\]|\\\\.)*?(\\6)))))?\\s*(\\])'
         'name': 'meta.attribute-selector.css'
       }
+      {
+        'match': '(?<![\\w-])[a-z][\\w&&[^A-Z]]*+-[\\w-&&[^A-Z]]++'
```
Comment (50Wliu) [inline grammars/css.cson]: Why the possessive matching at the end?

File: spec/css-spec.coffee [hunk 2]
```diff
@@ -95,89 +113,54 @@ describe 'CSS grammar', ->
...
         expect(tokens[1]).toEqual value: '::', scopes: ['source.css', 'meta.selector.css', 'entity.other.attribute-name.pseudo-class.css', 'punctuation.definition.entity.css']
         expect(tokens[2]).toEqual value: 'shadow', scopes: ['source.css', 'meta.selector.css', 'entity.other.attribute-name.pseudo-class.css']
 
-      it 'tokenizes them with class selectors', ->
-        {tokens} = grammar.tokenizeLine 'very-custom.class { color: red; }'
-        expect(tokens[0]).toEqual value: 'very-custom', scopes: ['source.css', 'meta.selector.css', 'entity.name.tag.custom.css']
-        expect(tokens[1]).toEqual value: '.', scopes: ['source.css', 'meta.selector.css', 'entity.other.attribute-name.class.css', 'punctuation.definition.entity.css']
-        expect(tokens[2]).toEqual value: 'class', scopes: ['source.css', 'meta.selector.css', 'entity.other.attribute-name.class.css']
-
-      it 'does not confuse them with properties', ->
```
Comment (50Wliu) [inline spec/css-spec.coffee]: Why was this test removed?

File: grammars/css.cson (hunk 1 above)
Comment (esdoppio) [inline grammars/css.cson]: 👍  good point

File: spec/css-spec.coffee (hunk 2 above)
Comment (esdoppio) [inline spec/css-spec.coffee]: Not entirely removed, covered by LINE 163 now.

Comment (50Wliu) [inline spec/css-spec.coffee]: What about the extra whitespace though?

Comment (esdoppio) [inline spec/css-spec.coffee]: I guess it should be fine. I mean, look at the new pattern of custom tag, whether it is followed by spaces or colon is not really relevant. And I think this test is not quite necessary in the first place unless people try to include the `#selector` entry in `#rule-list` and put it before the pattern of properties, it won't be confused with the properties. Besides, the property lists suite already has a test for extraneous spaces. 😉

**Pull request merged**